internal/pkg/source/source.go

```diff
@@ -462,6 +465,52 @@ func isSourceType(c classifier, t types.Type) bool {
 	}
 }
 
+// A type "can point" if it is itself a pointer or pointer-like type, or it contains
+// a pointer or pointer-like type.
+func canPoint(t types.Type) bool {
+	switch tt := t.(type) {
+	// These types can point.
+	// Chan, Map and Slice are reference-like collections.
+	// Interfaces can hold pointers.
+	case *types.Chan, *types.Interface, *types.Map, *types.Pointer, *types.Slice:
+		return true
+
+	case *types.Basic:
+		return tt.Kind() == types.UnsafePointer
+
+	// These types cannot point.
+	// Signatures represent functions, which for our purpose aren't pointers.
+	// Tuples only occur within functions.
+	case *types.Signature, *types.Tuple:
+		return false
+
```

Comment:
This feels out of scope for both this function and this PR.  `Signature`s are pointer-like, and I would expect a method named `canPoint` to reflect that.  Whether or not they can be meaningfully tainted is another question, and one that feels like it should be owned by `Source.visit`.

Reply:
That does sound like a better separation of work. I'll address `Signatures` and `Tuples` in a future PR, then.